### ISSUE TITLE
Need a way to respond to logout similar to current respondToLoginSucceed

### ISSUE THREAD

**Issue by aadamowski** [bnoguchi/everyauth#218]
Hi!

This is sort of a follow-up issue to [issue 41](/bnoguchi/everyauth/issues/41).

The example use case is the same. I want to redirect the user to the original url after login, so I'm setting the original url in req.session.redirectTo.

1st, the example in [issue 41](/bnoguchi/everyauth/issues/41) is incomplete - before redirecting the user on login success, one should unset the redirect url so that the redirection takes place only once and doesn't litter session data, so instead:

``` javascript
res.writeHead(303, {'Location': data.session.redirectTo});
```

it should be:

``` javascript
var redirectTo = data.session.redirectTo;
delete data.session.redirectTo;
res.writeHead(303, {'Location': redirectTo});
```

2nd, and this is the subject of this issue, it should be also possible to clean up the session stuff on logout, so get rid of any  `session.redirectTo` that would be set when visiting pages when logged in (who would have guarded all those places with conditionals that depend on user deauthentication?).

Otherwise, when the given user tries to log back in, he may get redirected to a completely unexpected url.

The problem is, there's currently no extension point for logout:

``` javascript
everyModule
  .configurable({
      moduleTimeout: 'how long to wait per step ' +
        'before timing out and invoking any ' +
        'timeout callbacks'
    , moduleErrback: 'THE error callback that is invoked' +
        'any time an error occurs in the module; defaults to `throw` wrapper'
    , logoutRedirectPath: 'where to redirect the app upon logging out'
    , findUserById: 'function for fetching a user by his/her id -- used to assign to `req.user` - function (userId, callback) where function callback (err, user)'
  })  
  .get('logoutPath')
    .step('handleLogout')
      .accepts('req res')
      .promises(null)
  .logoutPath('/logout')
  .handleLogout( function (req, res) {
    req.logout();
    res.writeHead(303, { 'Location': this.logoutRedirectPath() });
    res.end();
  })
  .logoutRedirectPath('/');
```

The only extension function available, `logoutRedirectPath`, doesn't receive access to `req` or `req.session`.
